src/VisualStudio/Xaml/Impl/Features/Diagnostics/IXamlPullDiagnosticService.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Features.Diagnostics
+{
+    internal interface IXamlPullDiagnosticService : ILanguageService
+    {
+        Task<XamlDiagnosticReport> GetDiagnosticReportAsync(TextDocument document, string? resultId, CancellationToken cancellationToken);
```

Comment:
```suggestion
        Task<XamlDiagnosticReport> GetDiagnosticReportAsync(TextDocument document, string? previousResultId, CancellationToken cancellationToken);
```

Consider doc'ing this, esp. 'previousResultId'

Reply:
Done.